src/components/Photos/index.js

```diff
@@ -11,12 +11,149 @@ import { wallpaperPhotos } from './imgCategories/Wallpapers';
+	var c = Object.keys(categories)
+	c.shift()
+	var dates = []
+	c.map(category => (
+	    categories[category].map(img => (
+		dates.push({
+		    'imgname':img.imgname,
+		    'date':Date.parse(img.date),
+		    'category':category,
+		    'name':img.name,
+		    'desc':img.desc,
+		    'credits':img.credits,
+		})		
+	    ))))
+	if (sort === "Date Desc.") {dates.sort((a, b) => a.date - b.date)}
+	else {dates.sort((a, b) => b.date - a.date)}
+	return dates.map(img => (
+		<React.Fragment>
+		<div id="myModal" className="modal">
+		<span
+	        className="close"
+	        onClick={() => document.getElementById("myModal").style.display = 'none'}
+		>&times;</span>
+		<div id="image-title"></div>
+		<div style={{display: "flex"}}>
+		<img className="modal-content" id="big-image"/>
+		<p id="image-credits"></p>
+		<div id="image-desc"></div>
+		</div>
+		</div>
+                <img
+	        id="myImg"
+                className="imag"
+	        onClick={() => {
+		    document.getElementById("big-image").setAttribute('src', require('./imgCategories/' + img.category + '/' + img.imgname))
+		    document.getElementById("image-desc").innerHTML = img.desc
+		    document.getElementById("image-title").innerHTML = img.name
+		    document.getElementById("image-credits").innerHTML = "Photo Credits: " + img.credits
+		 document.getElementById("myModal").style.display = 'block'	
+		}}
+	    src={require('./imgCategories/' + img.category + '/' + img.imgname)} />
+                
+		</React.Fragment>		    
+	))
+    }
+    else {
+	var one_category = categories[category]
+	if (sort === "Date Desc.") {one_category.sort((a, b) => Date.parse(a.date) - Date.parse(b.date))}
+	else {one_category.sort((a, b) => Date.parse(b.date) - Date.parse(a.date))}
+	return one_category.map(img => (
+	    <React.Fragment>
+		<div id="myModal" className="modal">
+		<span
+	        className="close"
+	        onClick={() => document.getElementById("myModal").style.display = 'none'}
+		>&times;</span>
+		<div id="image-title"></div>
+		<div style={{display: "flex"}}>
+		<img className="modal-content" id="big-image" />
+		<p id="image-credits"></p>
+		<div id="image-desc"></div>
+		<a id="download" download></a>
+		</div>
+		</div>
+                <img
+            className="imag"
+	    onClick={() => {
+		document.getElementById("big-image").setAttribute('src', require('./imgCategories/' + category + '/' + img.imgname))
+		document.getElementById("image-desc").innerHTML = img.desc
+		document.getElementById("image-title").innerHTML = img.name
+		document.getElementById("download").innerHTML = 'Download'
+		document.getElementById("download").setAttribute("href", './imgCategories/' + category + '/' + img.imgname)
+		document.getElementById("image-credits").innerHTML = "Photo Credits: " + img.credits
+		document.getElementById("myModal").style.display = 'block'	
+	    }}
+	    src={require('./imgCategories/' + category + '/' + img.imgname)} />
+	    </React.Fragment>
```

Comment:
Create another component to encapsulate all of this and then import it into this file.
You can pass information into another component like this:
`<AnotherComponent  detail={something} anotherthing={somethingelse} />`
and then you can access those pieces of information inside of your `AnotherComponent` code
by using `this.props.detail` or `this.props.anotherthing`.
This will make the code a lot more cleaner since you are repeating the same thing on top.